lib/actions-oclif/note.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright 2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import { stripIndent } from 'common-tags';
+import Command from '../command';
+import * as cf from '../utils/common-flags';
+import { getBalenaSdk } from '../utils/lazy';
+import { CommandHelp } from '../utils/oclif-utils';
+
+interface FlagsDef {
+	device: string; // device UUID
+	help: void;
+}
+
+interface ArgsDef {
+	note: string;
+}
+
+export default class NoteCmd extends Command {
+	public static description = stripIndent`
+		Set a device note.
+
+		Use this command to set or update a device note.
+
+		If note command isn't passed, the tool attempts to read from \`stdin\`.
+
+		To view the notes, use $ balena device <uuid>.
+`;
+	public static examples = [
+		'$ balena note "My useful note" --device 7cf02a6',
+		'$ cat note.txt | balena note --device 7cf02a6',
+	];
+
+	public static args = [
+		{
+			name: 'note',
+			required: true,
```

Comment:
I was going to say that the `note` args is not required (`required: false`) because when not provided, the command reads from stdin. However I see in the old implementation:
```
		if (_.isEmpty(params.note)) {	
			const { exitWithExpectedError } = await import('../utils/patterns');	
			exitWithExpectedError('Missing note content');	
		}
```

So the command never actually attempts reading from stdin, right? In which case, the argument is indeed required, and what we need to change is the help output, which should not mention reading from stdin...

Reply:
It's supposed to accept piped input. I though I had tested this, but in tests just now it seems it's not working.  Will look into it.

Reply:
Yeah seems oclif doesn't support piped input, but there is mention of a workaround here (trying to figure out what he meant)

https://github.com/oclif/oclif/issues/245#issuecomment-576469479

Reply:
Before this PR, it was not working with Capitano either:
```
$ balena note --device 123
Missing note content
```
Also as per code snippet I quoted above. Therefore, I'm happy with simply amending the help output to remove mention of stdin.

Reply:
@pdcastro these currently work with capitano though:
```
$ cat test.txt | balena note -d 123

$ bin/balena note -d 123 < test.txt
```

Reply:
Oh okay. In this case we need to get it working with oclif as well, or leave the Capitano implementation for now. (We could also drop support from reading from stdin in a major CLI release like v12.) Someone has replied to your comment in the oclif issue, maybe we can implement something like that. We'd also need to test with cmd.exe, PowerShell and bash... to at least maintain what currently works.

Reply:
I've added piped input support, using the same library that capitano uses.  It should behave exactly the same whatever the platform, but testing now.

Reply:
Now tested working on Windows cmd, powershell, msys